src/scheduling/cost_models/simulated_dfs.h

```diff
@@ -0,0 +1,58 @@
+#ifndef SRC_SCHEDULING_COST_MODELS_SIMULATED_DFS_H_
+#define SRC_SCHEDULING_COST_MODELS_SIMULATED_DFS_H_
+
+#include <utility>
+#include <unordered_set>
+#include <unordered_map>
+#include <queue>
+#include <list>
+#include <random>
+
+#include "base/types.h"
+#include "base/resource_topology_node_desc.pb.h"
+#include "scheduling/cost_models/google_block_distribution.h"
+
+namespace firmament {
+
+class SimulatedDFS {
+public:
+	typedef uint64_t FileID_t;
+	typedef uint32_t NumBlocks_t;
+
+	SimulatedDFS(uint64_t num_machines, NumBlocks_t blocks_per_machine,
+	    uint32_t replication_factor, GoogleBlockDistribution *block_distribution,
+	    uint64_t random_seed);
+	virtual ~SimulatedDFS();
+
+	void addMachine(ResourceID_t machine);
+	void removeMachine(ResourceID_t machine);
+
+	NumBlocks_t getNumBlocks(FileID_t file) const {
+	  return files[file];
+	}
+	const ResourceSet_t getMachines(FileID_t file) const;
+	const std::unordered_set<FileID_t> sampleFiles(NumBlocks_t num_blocks,
```

Comment:
I *think* it randomly picks input files covers num_blocks DFS blocks for a job. @AdamGleave should be able to confirm.